packages/indexer/src/sinks/drizzle/delete.ts

```diff
@@ -0,0 +1,46 @@
+import type { Cursor } from "@apibara/protocol";
+import {
+  type ExtractTablesWithRelations,
+  type SQL,
+  type TablesRelationalConfig,
+  sql,
+} from "drizzle-orm";
+import type {
+  PgQueryResultHKT,
+  PgTable,
+  PgTransaction,
+  PgUpdateBase,
+} from "drizzle-orm/pg-core";
+
+export class DrizzleSinkDelete<
+  TTable extends PgTable,
+  TQueryResult extends PgQueryResultHKT,
+  TFullSchema extends Record<string, unknown> = Record<string, never>,
+  TSchema extends
+    TablesRelationalConfig = ExtractTablesWithRelations<TFullSchema>,
+> {
+  constructor(
+    private db: PgTransaction<TQueryResult, TFullSchema, TSchema>,
+    private table: TTable,
+    private endCursor?: Cursor,
+  ) {}
+
+  //@ts-ignore
+  where(where: SQL): Omit<
+    // for type safety used PgUpdateBase instead of PgDeleteBase
+    PgUpdateBase<TTable, TQueryResult, undefined, false, "where">,
+    "where"
+  > {
+    const updatedRows = this.db
+      .update(this.table)
+      .set({
+        // @ts-ignore
+        _cursor: sql`int8range(lower(_cursor), ${this.endCursor?.orderKey!}, '[]')`,
```

Comment:
Why not use `getDrizzleCursor`?

Reply:
> Why not use `getDrizzleCursor`?

i was unable to get the current value of the cursor for the rows. Normally we do something like this as per drizzle docs right. 

```
await db.update(users)
  .set({ name: 'Mr. Dan' })
  .where(eq(users.name, 'Dan'));
```

but as the `getDrizzleCursor` takes `bigint` as args, this wont work with drizzle `set` method logic. Do you have any alternative and more proper solution to this maybe?

Reply:
True. I think it's fine then since the alternative may end up being more complex than this.